app/screens/global_threads/threads_list/file_card/file_card.tsx

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React, {useCallback, useEffect, useState} from 'react';
+import {Text, View} from 'react-native';
+
+import {buildFilePreviewUrl, buildFileThumbnailUrl} from '@actions/remote/file';
+import FileIcon from '@components/files/file_icon';
+import ProgressiveImage from '@components/progressive_image';
+import {useServerUrl} from '@context/server';
+import {getFormat, getFormattedFileSize} from '@utils/file';
+import {FileTypes} from '@utils/file/constants';
+
+import {getStyleSheet} from './styles';
+
+import type FileModel from '@typings/database/models/servers/file';
+import type PostModel from '@typings/database/models/servers/post';
+
+type CardProps = {
+    post: PostModel;
+    theme: Theme;
+};
+
+export const FileCard: React.FC<CardProps> = ({post, theme}) => {
+    const [files, setFiles] = useState<FileModel[]>([]);
+    const [failed, setFailed] = useState(false);
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    useEffect(() => {
+        const fetchFiles = async () => {
+            const fetchedFiles = await post.files.fetch();
+            setFiles(fetchedFiles);
+        };
+
+        fetchFiles();
+    }, [post]);
```

Comment:
This works, but it won't react to changes in the files until the post changes. In practice, I think that will never happen, but just in case we should do this "the correct way".

For that you should add a watermelon db HOC, like the one in `app/components/common_post_options/copy_permalink_option/index.tsx`, and return there something like `post.files.observe()`. If you need more indications for this, let me know.